sdk/python/feast/infra/offline_stores/offline_store.py

```diff
@@ -73,6 +87,24 @@ def to_arrow(self) -> pyarrow.Table:
             )
         return pyarrow.Table.from_pandas(features_df)
 
+    @abstractmethod
+    def persist(self, storage: SavedDatasetStorage) -> "RetrievalJob":
```

Comment:
why do you need to do a forward reference type check? doesn't seem like there's a circular dependency?

Reply:
the idea is that persist returns new RetrievalJob, it could be modified self or completely new job (that would wrap new query). And the result of persist is what returned from `get_historical_features`

Reply:
makes sense. See my other comment for further thoughts.